ssz-rs/src/merkleization/mod.rs

```diff
@@ -288,27 +309,38 @@ mod tests {
         debug_assert!(chunks.len() % BYTES_PER_CHUNK == 0);
         debug_assert!(leaf_count.next_power_of_two() == leaf_count);
 
+        // SAFETY: checked subtraction is unnecessary,
+        // as leaf_count != 0 (0.next_power_of_two() == 1); qed
         let node_count = 2 * leaf_count - 1;
+        // SAFETY: checked subtraction is unnecessary, as node_count >= leaf_count; qed
         let interior_count = node_count - leaf_count;
         let leaf_start = interior_count * BYTES_PER_CHUNK;
 
         let mut hasher = Sha256::new();
         let mut buffer = vec![0u8; node_count * BYTES_PER_CHUNK];
         buffer[leaf_start..leaf_start + chunks.len()].copy_from_slice(chunks);
-        let zero_chunk = [0u8; 32];
+        let zero_chunk = [0u8; BYTES_PER_CHUNK];
         for i in chunks.len()..leaf_count {
             let start = leaf_start + (i * BYTES_PER_CHUNK);
             let end = leaf_start + (i + 1) * BYTES_PER_CHUNK;
             buffer[start..end].copy_from_slice(&zero_chunk);
         }
```

Comment:
I don't think the bounds on this iterator are correct `chunks.len()..leaf_count` and it was fine only bc we are just writing zeros to memory that was already zero (initialized in `buffer`)

I deleted it anyway bc it didn't seem to be doing anything